cpp/include/cuvs/neighbors/cagra.hpp

```diff
@@ -169,6 +169,9 @@ struct search_params : cuvs::neighbors::search_params {
   uint32_t num_random_samplings = 1;
   /** Bit mask used for initial random seed node selection. */
   uint64_t rand_xor_mask = 0x128394;
+
+  /** Whether to use the persistent version of the kernel (only SINGLE_CTA is supported a.t.m.) */
+  bool persistent = false;
```

Comment:
By accident I have posted one of my questions to the RAFT version of this PR. For reference I post the question and the answer here again.

Is this the only parameter that needs to be controlled? What about the temporary buffer size (queue for queries, or max number of threads)?


@[achirkin](https://github.com/achirkin) achirkin [3 days ago]:(https://github.com/rapidsai/raft/pull/2316#discussion_r1675514926)
The buffer sizes are compile-time constants in the current design. I'd like to move some of the new constants to the parameters, but that conflicts with the implicit mechanics of running the kernel. They are not search parameters, but the "runner" parameters and should not be changed across calls to the kernel.
So this brings up again the question whether we want to have implicit vs explicit kernel runner.